Content.Server/Internals/InternalsSuit.cs

```diff
@@ -0,0 +1,133 @@
+            => SendToggleEvent(uid, component, new ToggleInternalsEvent(false, true, GetActions(uid)));
+
+        public void OnToggleInternals(EntityUid uid, InternalsSuitComponent component, ToggleInternalsEvent args)
+            => SendToggleEvent(uid, component, args);
+
+        public void SendToggleEvent(EntityUid uid, InternalsSuitComponent component, ToggleInternalsEvent args)
+        {
+            if (component.GasTankPresent)
+                RaiseLocalEvent(component.GasTankContainer.ContainedEntity!.Uid, args);
+        }
+
+        private void OnExamined(EntityUid uid, InternalsSuitComponent component, ExaminedEvent args)
+        {
+            if (!args.IsInDetailsRange)
+                return;
+
+            args.PushMarkup(Loc.GetString("generic-error-no-hands", ("present", component.GasTankPresent)));
+
+            // TODO: add a pressure gauge / approximate one?
+            if (
+                component.GasTankPresent
+                && component.GasTankContainer.ContainedEntity!.TryGetComponent(out InternalsProviderComponent? provider)
+            )
+                Get<InternalsProviderSystem>().OnSuitExamined(uid, component, provider, args);
+        }
+
+        private async void OnInteractUsing(EntityUid uid, InternalsSuitComponent component, InteractUsingEvent args)
+        {
+            var user = args.User;
+
+            if (!user.TryGetComponent(out SharedHandsComponent? hands))
+            {
+                component.Owner.PopupMessage(user, Loc.GetString("generic-error-no-hands")); // Do we even need to check that...? Why?
+                return;
+            }
+
+            if (args.Used.HasComponent<InternalsProviderComponent>())
+            {
+                if (component.GasTankPresent)
+                    component.Owner.PopupMessage(user, Loc.GetString("internals-suit-already-occupied"));
+                else if (hands.TryPutHandIntoContainer(hands.ActiveHand!, component.GasTankContainer))
+                    component.Owner.PopupMessage(user, Loc.GetString("internals-suit-attached-success", ("attachedTo", args.Target.Name)));
+            }
+            else if (await Get<ToolSystem>().UseTool(args.Used.Uid, args.User.Uid, component.Owner.Uid, 0f, 0.25f, component._screwingQuality))
+            {
+                if (!component.GasTankPresent)
+                {
+                    args.Target.PopupMessage(user, Loc.GetString("internals-suit-already-empty"));
+                    return;
+                }
+                var entity = component.GasTankContainer.ContainedEntity!;
+                if (component.GasTankContainer.Remove(entity!))
+                    if (entity.TryGetComponent(out SharedItemComponent? item))
+                        hands.PutInHand(item);
+            }
+            else
+                component.Owner.PopupMessage(user, Loc.GetString("internals-suit-invalid-type"));
```

Comment:
Do we not have tool events? We're trying to remove the do_after asyncs slowly.

Also EntityManager again, + popupsystem for the popups.